### DISCUSSION TITLE
'Snap Engine'

### DISCUSSION THREAD

**Discussion by computer-droid** [leozide/leocad#690] [Ideas]
**You should make a *'snap engine'* - an engine which allows people to 'snap' or 'attach' LEGO pieces to each other.**
When I use LeoCAD, it's very hard to attach Lego pieces (especially LEGO Technic pins with LEGO Brick with hole).
LDD (Lego Digital Designer) had that feature, but unfortunately, they ended support.

**Comment by nathaneltitane** [comment]
this proposal makes sense:
if we are able to map stud locations for primitive and texture mapping (logo) then there must be a way to orient parts using preselection (blue) and having the newly inserted part orient its base to whatever stud it can be dragged around or on

**Comment by bafly** [comment]
++ snap engine! please!
